### PR TITLE
Fix StatelessComponent interface

### PR DESCRIPTION
**Pull request by Nitive**
Empty object is passed to StatelessComponent even if we pass no props.
Example:

``` js
function Test(props) {
  return <div>props: {JSON.stringify(props)}</div>
}

React.render(<Test />, document.body)
```

will render into `props: {}`
[jsfiddle](https://jsfiddle.net/LL25c28c/)

We need to fix this because otherwise we have error:
`Property 'propName' does not exist on type 'IntrinsicAttributes & (IMyInterface | undefined)'.`
for that code:

``` js
const MyComponent: React.SFC<IMyInterface> = ({ propName }: IMyInterface) => {
  // ...
}
```

### CODE REVIEW

**Comment by vvakame** [comment]
*react/react.d.ts*

to authors (@pspeter3 @vsiao AssureSign (account can't be detected) Microsoft (account can't be detected)). Could you review this PR?
:+1: or :-1:?

Checklist

* [ ] pass the Travis CI test?

**Comment by vsiao** [comment]
Is there a test that you can add for this?

**Comment by Nitive** [comment]
Found an issue about that problem #10388 
Looks like the problem only for TypeScript 2 with `"strictNullChecks": true` and only when JSX is used.

It would be test but we need set up typescript@2 and jsx first
```typescript
var StatelessComponent3: React.SFC<SCProps> = ({ foo }: SCProps) => {
    return React.DOM.div(null, foo);
};
StatelessComponent3.displayName = "StatelessComponent3";
StatelessComponent3.defaultProps = {
    foo: 42
};

<StatelessComponent3 />; // should not throw
```

**Comment by vvakame** [comment]
Can I merge this PR?

**Comment by vsiao** [comment]
@Nitive, if you could add that test anyway that would be great. Otherwise this PR is good to go 👍

**Comment by Nitive** [comment]
@vsiao If I will add it, tests will fail with syntax error

**Comment by vsiao** [comment]
What's the syntax error? It's fine if you can't.

**Comment by Nitive** [comment]
@vsiao I do not know how configure jsx in this project. Adding `tsconfig.json` in `react/` did not help.

**Comment by vsiao** [comment]
Oh, I see. Can you try adding a `react-tsx-tests.tsx` file? The `.tsx` extension is probably necessary.

**Comment by Nitive** [comment]
@vsiao Added

File: react/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+    "compilerOptions": {
+        "jsx": "react",
+        "module": "commonjs",
+        "target": "es3",
+        "strictNullChecks": true,
+        "outDir": "."
+    }
+}
```
Comment (vsiao) [inline react/tsconfig.json]: What is this file for?

Comment (Nitive) [inline react/tsconfig.json]: O, we does not need any configuration for jsx, just .tsx extension. I deleted the file.

**Comment by vsiao** [comment]
great, thanks! 👍

**Pull request merged**